Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -6,15 +6,34 @@
 
 class IntegrationCommandTests < Homebrew::TestCase
   def setup
-    @formula_files = []
     @cmd_id_index = 0 # Assign unique IDs to invocations of `cmd_output`.
     (HOMEBREW_PREFIX/"bin").mkpath
     FileUtils.touch HOMEBREW_PREFIX/"bin/brew"
   end
 
   def teardown
-    (HOMEBREW_PREFIX/"bin").rmtree
-    @formula_files.each(&:unlink)
+    coretap = CoreTap.new
+    paths_to_delete = []
+    paths_to_delete += HOMEBREW_CELLAR.children
+    paths_to_delete += HOMEBREW_CACHE.children
+    paths_to_delete += HOMEBREW_LOCK_DIR.children
+    paths_to_delete += HOMEBREW_LOGS.children
+    paths_to_delete += [
+      HOMEBREW_PREFIX/"bin",
+      HOMEBREW_PREFIX/"share",
+      HOMEBREW_PREFIX/"opt",
+      HOMEBREW_LIBRARY/"LinkedKegs",
+      HOMEBREW_LIBRARY/"Taps/caskroom",
+      HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-bundle",
+      HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-foo",
+      HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-services",
+      HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-shallow",
+      HOMEBREW_REPOSITORY/".git",
+      coretap.alias_dir,
+      coretap.path/".git",
+    ]
+    paths_to_delete += coretap.formula_dir.children
```

Comment:
Just an idea, but you could also do the following to avoid constantly repeating `paths_to_delete`:

```ruby
paths_to_delete = [
  HOMEBREW_CELLAR.children,
  HOMEBREW_CACHE.children,
  HOMEBREW_LOCK_DIR.children,
  HOMEBREW_LOGS.children,
  HOMEBREW_PREFIX/"bin",
  HOMEBREW_PREFIX/"share",
  HOMEBREW_PREFIX/"opt",
  HOMEBREW_LIBRARY/"LinkedKegs",
  HOMEBREW_LIBRARY/"Taps/caskroom",
  HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-bundle",
  HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-foo",
  HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-services",
  HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-shallow",
  HOMEBREW_REPOSITORY/".git",
  coretap.path/".git",
  coretap.alias_dir,
  coretap.formula_dir.children,
].flatten
```